act/plotting/plot.py

```diff
@@ -278,3 +279,126 @@ def add_colorbar(self, mappable, title=None, subplot_index=(0,), pad=None,
         self.cbs.append(cbar)
 
         return cbar
+
+    def group_by(self, units):
+        """
+        Group the Display by specific units of time.
+
+        Parameters
+        ----------
+        units: str
+            One of: 'year', 'month', 'day', 'hour', 'minute', 'second'.
+            Group the plot by this unit of time (year, month, etc.)
+        Returns
+        -------
+        groupby: act.plotting.DisplayGroupby
+            The DisplayGroupby object to be retuned.
+        """
+        return DisplayGroupby(self, units)
+
+
+class DisplayGroupby(object):
+    def __init__(self, display, units):
+        """
+
+        Parameters
+        ----------
+        display: Display
+            The Display object to group by time.
+        units: str
+            The time units to group by. Can be one of:
+            'year', 'month', 'day', 'hour', 'minute', 'second'
+        """
+        self.display = display
+        self._groupby = {}
+        num_groups = 0
+        datastreams = list(display._obj.keys())
+        for key in datastreams:
+            self._groupby[key] = display._obj[key].groupby('time.%s' % units)
+            num_groups = max([num_groups, len(self._groupby[key])])
+
+    def plot_group(self, func_name, dsname, **kwargs):
```

Comment:
Is there anyway we can automatically fill the dsname to be None?

Reply:
Done in new commit!